google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.ads.googleads.examples.remarketing;
+
+import com.beust.jcommander.Parameter;
+import com.google.ads.googleads.examples.utils.ArgumentNames;
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v8.common.UserIdentifier;
+import com.google.ads.googleads.v8.enums.UserIdentifierSourceEnum.UserIdentifierSource;
+import com.google.ads.googleads.v8.errors.GoogleAdsError;
+import com.google.ads.googleads.v8.errors.GoogleAdsException;
+import com.google.ads.googleads.v8.services.ClickConversion;
+import com.google.ads.googleads.v8.services.ClickConversionResult;
+import com.google.ads.googleads.v8.services.ConversionUploadServiceClient;
+import com.google.ads.googleads.v8.services.UploadClickConversionsRequest;
+import com.google.ads.googleads.v8.services.UploadClickConversionsResponse;
+import com.google.ads.googleads.v8.utils.ResourceNames;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/** Uploads a conversion using hashed email address instead of GCLID. */
+public class UploadConversionWithIdentifiers {
+  private static class UploadConversionWithIdentifiersParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private long customerId;
+
+    @Parameter(names = ArgumentNames.CONVERSION_ACTION_ID, required = true)
+    private long conversionActionId;
+
+    @Parameter(names = ArgumentNames.EMAIL_ADDRESS, required = true)
+    private String emailAddress;
+
+    @Parameter(names = ArgumentNames.ORDER_ID, required = false)
```

Comment:
Nit: I'd change the order to make this last to be consistent with the params order below.